android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+Example of starting a new Activity and passing data to it, notice the Activity's constructor is not called:
+```
+Intent intent = new Intent(this, ActivityTwo.class);
+intent.putExtra("EXTRA_KEY_COUNT", 1);
+startActivity(intent);
+```
+
+In iOS, the presenting VC can have a direct reference to the presented VC, and often times the presenting VC directly calls the presented VC’s initializer and pass any parameters it needs. This also means that it can call the new VC's methods and modify its properties before showing it.
+
+## Layout building
+iOS layouts are done using [Auto Layout](https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/AutolayoutPG/index.html), which dynamically calculates the size and position of views based on a set of constraints. All views implicitly uses Auto Layout, so there's no explicit declaration of what layout is being used like Android.\
+iOS layouts are usually built through the XCode interface builder, though building them programmatically with Swift is not uncommon. Editing through xml is almost never done other than when resolving merge conflicts. xibs are preferred over storyboards for larger projects as it is less prone to merge conflicts and is more flexible (especially in regards to MVVM).
+
+Modern Android layouts are typically done using [ConstraintLayout](https://developer.android.com/reference/android/support/constraint/ConstraintLayout), which is very similar to iOS's Auto Layout. [LinearLayout](https://developer.android.com/guide/topics/ui/layout/linear) and [FrameLayout](https://developer.android.com/reference/android/widget/FrameLayout) are occasionally used for simpler views, and [RelativeLayouts](https://developer.android.com/guide/topics/ui/layout/relative) were used on older projects as it is deprecated by ConstraintLayout. A Layout is just a container that dictates how its children are positioned. Each layout file will have one of these layouts as the top level container, and it is possible (and quite common) for a layout to contain another layout.\
+Historically, layouts are generally built by writing xml instead of using the interface builder, partly due to Android Studio's interface builder being terrible/inadequate. The interface builder has improved quite a lot recently with the introduction of ConstraintLayout, so it has more usage nowadays. However, building the layout via by xml might still be more preferable to a lot of people since it offers more control, and Android Studio can help autocomplete many of the fields.
```

Comment:
The layout preview view also serves as a sort of hybrid solution here, as you can edit the xml directly but still see the visual.